### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by RaoAkif**
This Pull Request follows the following requirements:

#### Interactions
- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (catagories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

#### Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.

#### Technical requirements

- Use Postgres as your database.
- Use devise for authentication.
- Validate all user input to make sure that anyone with bad intentions cannot compromise your app.
- Use a view template engine of your choice (.erb, .slim, .haml).
- The project is deployed and accessible online.

### CODE REVIEW

File: app/views/categories/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+        <div class="category-details">
+          <%= image_tag "#{category.icon}", style: 'height:50px; width:50px;' %>
+          <div><%= category.name %></div>
+        </div>
+        <div>$<%= category.total_amount %></div>
+      </div>
+    <% end %>
+  <% end %>
+  <div><%= link_to "Add a Category", new_category_path, class: "btn login-btn bottom-button" %></div>
+</div>
```
Comment (rotimiazeez) [inline app/views/categories/index.html.erb:16]: - Please note that when trying to select an Icon for the category the below error was raised
Screenshot from 2022-08-06 09-24-32
Kindly fix this 👍🏽

File: README.md [hunk 2]
```diff
@@ -21,7 +21,11 @@ Finally, the `Transactions Page` is where you can see the details from all the t
 
 ## Live Demo
 
-[Live Demo Link](https://livedemo.com)
+[Live Demo Link](https://www.loom.com/share/09153510f56b4d3cadc135b31e11d3b2)
+
+## Deployment
+
```
Comment (rotimiazeez) [inline README.md:27]: - Make sure your README contains an appropriate credit to Gregoire Vella on Behance.

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi @RaoAkif  👏🏽, 
## Great job you did here 💯 
- Good UI 😃  
- Descriptive PR 💯 
- Good commit messages 🏷️
- Correct git-flow followed 👍🏽 
- No linter errors ✔️
- All tests are passing 🥇 
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/categories/index.html.erb (hunk 1 above)
Comment (RaoAkif) [reply app/views/categories/index.html.erb:16]: @rotimiazeez just give the lcon url for example take it from seeds file.
Thats it.

File: README.md (hunk 2 above)
Comment (RaoAkif) [reply README.md:27]: OK

File: app/views/categories/index.html.erb (hunk 1 above)
Comment (RaoAkif) [reply app/views/categories/index.html.erb:16]: `https://img.icons8.com/color/480/000000/shopping-cart-loaded.png`

**Review by aimemalaika — APPROVED** [review]
Hi @RaoAkif , 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

133591850-fdcb7dd3-dd7a-4274-81e7-0e59434148f7

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**